electron/src/js/lib/SingleSignOn.ts

```diff
@@ -0,0 +1,318 @@
+      ...this.windowOptions,
+      backgroundColor: '#FFFFFF',
+      height: this.windowOptions.height || 600,
+      modal: false,
+      parent: this.mainBrowserWindow,
+      resizable: false,
+      titleBarStyle: 'hiddenInset',
+      webPreferences: {
+        ...this.windowOptions.webPreferences,
+        allowRunningInsecureContent: false,
+        backgroundThrottling: false,
+        contextIsolation: true,
+        devTools: true,
+        disableBlinkFeatures: '',
+        enableBlinkFeatures: '',
+        experimentalCanvasFeatures: false,
+        experimentalFeatures: false,
+        images: true,
+        javascript: true,
+        nativeWindowOpen: false,
+        nodeIntegration: false,
+        nodeIntegrationInWorker: false,
+        offscreen: false,
+        partition: '',
+        plugins: false,
+        preload: SingleSignOn.PRELOAD_SSO_JS,
+        sandbox: true,
+        scrollBounce: true,
+        session: this.session,
+        textAreasAreResizable: false,
+        webSecurity: true,
+        webgl: false,
+        webviewTag: false,
+      },
+      width: this.windowOptions.width || 480,
+    });
+
+    (<any>this.senderEvent).newGuest = SingleSignOnLoginWindow;
+
+    SingleSignOnLoginWindow.once('closed', async () => {
+      if (this.session) {
+        await this.wipeSessionData();
+        await SingleSignOn.protocol.unregister(this.session);
+      }
+      this.session = undefined;
+    });
+
+    return SingleSignOnLoginWindow;
+  };
+
+  private readonly finalizeLogin = async (type: string): Promise<void> => {
+    if (type === SingleSignOn.RESPONSE_TYPES.AUTH_SUCCESS) {
+      if (!this.session) {
+        await this.dispatchResponse(SingleSignOn.RESPONSE_TYPES.AUTH_ERROR_SESS_NOT_AVAILABLE);
+        return;
+      }
+
+      // Set cookies from ephemeral session to the default one
+      try {
+        await SingleSignOn.cookies.copy(this.session, this.mainSession, this.windowOriginUrl);
+      } catch (error) {
+        await this.dispatchResponse(SingleSignOn.RESPONSE_TYPES.AUTH_ERROR_COOKIE);
+        return;
+      }
+    }
+
+    await this.dispatchResponse(type);
+  };
+
+  private readonly dispatchResponse = async (type: string): Promise<void> => {
+    // Ensure guest window provided type is valid
+    if (/^[A-Z_]{1,255}$/g.test(type) === false) {
+      throw new Error('Invalid type detected, aborting.');
+    }
+
+    // Fake postMessage to the webview
+    await this.senderWebContents.executeJavaScript(
+      `window.dispatchEvent(new MessageEvent('message', {origin: '${
+        this.windowOriginUrl.origin
+      }', data: {type: '${type}'}, type: {isTrusted: true}}));`
+    );
+  };
+
+  private readonly wipeSessionData = () => {
+    return new Promise((resolve, reject) => {
+      if (this.session) {
+        this.session.clearStorageData({}, () => resolve());
+      } else {
+        resolve();
+      }
+    });
+  };
+}
+
+export {SingleSignOn};
```

Comment:
I guess this should be part of a different PR.. as this is for removing the google import.